packages/react-native/src/immutable-form/actions.js

```diff
@@ -38,8 +38,14 @@ export const submitForm = (formName: string) => (submit: (Object) => Promise<any
     const state = getState();
 
     submit(state).then(
-      res => dispatch(approvedForm(formName)(res)),
-      err => dispatch(errorForm(formName)(errorMessage(err))),
+      res => {
+        dispatch(clearForm(formName)());
+        dispatch(approvedForm(formName)(res));
```

Comment:
Actually when thinking more closely on this issue, we should maybe dispatch the clearForm action after the application navigates away from the form. It's a better user experience IMO and will make sure the state is coherent. Currently the form reducer resets the whole state to the initialState.
And @dotansimha I would not be concerned about the usage because trying  getting the password after form is submitted is definitely not a good behaviour.. 
If a user truly requires it, he could save on his own state by listening to the `submittedForm` action..